edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -829,6 +778,84 @@ where
     )
 }
 
+fn authenticate<MR>(
+    runtime: &MR,
+    req: &Request<Body>,
+) -> Box<dyn Future<Item = AuthId, Error = Error> + Send>
+where
+    MR: ModuleRuntime<Error = Error>,
+    <MR as ModuleRuntime>::ListFuture: 'static,
+    MR::Module: DockerModuleTop<Error = Error>,
+{
+    let pid = req
+        .extensions()
+        .get::<Pid>()
+        .cloned()
+        .unwrap_or_else(|| Pid::None);
+
+    let expected_module_id = req.extensions().get::<ModuleId>().cloned();
+
+    Box::new(match pid {
+        Pid::None => Either::A(future::ok(AuthId::None)),
+        Pid::Any => Either::A(future::ok(AuthId::Any)),
+        Pid::Value(pid) => {
+            Either::B(
+                // to authenticate request we need to determine whether given pid corresponds to
+                // any pid from any module. In order to do so, we are getting a list of all running
```

Comment:
The comment needs to be updated, since it doesn't do `docker top` for every module any more.

Reply:
Ok